lib/teleterm/gateway/gateway.go

```diff
@@ -58,11 +85,21 @@ func New(cfg Config) (*Gateway, error) {
 	}()
```

Comment:
Now this is interesting because if you look at the previous version of `gateway.New`, [this deferred function was "guarding" almost the whole `New` function](https://github.com/gravitational/teleport/blob/a41a4300708c3a72f2ed58ae9f1eeec89f5dcd87/lib/teleterm/gateway/gateway.go#L47-L56). So if `gateway.New` failed at any point after creating the listener, the context would be canceled and the listener closed.

In this version, this deferred function guards only `newListenerAndLocalProxy`. I think this is fine because both `New` and `SetLocalPort` don't currently have any calls that might fail after they call `newListenerAndLocalProxy`.

It might be tricky to keep this as invariant going forward though. OTOH, I had no idea how to write this and employing some callbacks/channels for this seemed too complicated for me here, though I might be wrong.